Analysis/src/TypeInfer.cpp

```diff
@@ -3640,6 +3640,9 @@ void TypeChecker::checkFunctionBody(const ScopePtr& scope, TypeId ty, const AstE
                 reportError(getEndLocation(function), FunctionExitsWithoutReturning{funTy->retTypes});
             }
         }
+
+        if (!currentModule->astTypes.find(&function))
```

Comment:
Why do we need the check?

Reply:
It was present in the checkExpr function, so I did the same here: https://github.com/Roblox/luau/blob/12c550202719b5deef932453728f5643bed23eaa/Analysis/src/TypeInfer.cpp#L1811

Can remove if needed

Reply:
Ah, odd. It's better to keep it for consistency for now then.